src/adapters/scene_direction.js

```diff
@@ -99,23 +100,42 @@ export default class SceneDirection {
 
       const markerOrigin = document.createElement('div')
       markerOrigin.className = this.vehicle === "walking" ? 'itinerary_marker_origin_walking' : 'itinerary_marker_origin'
-      this.markerOrigin = new Marker(markerOrigin)
+      this.markerOrigin = new Marker({
+        element: markerOrigin,
+        draggable: true
+      })
         .setLngLat(this.steps[0].maneuver.location)
         .addTo(this.map)
 
+      this.markerOrigin.on('dragend', (event) => this.refreshDirection(event, this, 'origin'))
+
       const markerDestination = document.createElement('div')
       markerDestination.className = 'itinerary_marker_destination'
-      this.markerDestination = new Marker(markerDestination)
+      this.markerDestination = new Marker({
+        element: markerDestination,
+        draggable: true
+      })
         .setLngLat(this.steps[this.steps.length - 1].maneuver.location)
         .addTo(this.map)
 
+      this.markerDestination.on('dragend', (event) => this.refreshDirection(event, this, 'destination'))
+
       let bbox = this.computeBBox(mainRoute);
       if(move !== false){
         fire('fit_map', bbox, layouts.ITINERARY)
       }
     }
   }
 
+  refreshDirection(event, _self, type) {
+    var originlngLat = this.markerOrigin.getLngLat();
+    var destinationlngLat = this.markerDestination.getLngLat();
+    _self.directionPanel.setInputValue(type, type === 'origin' ?
+      `${parseFloat(originlngLat.lat).toFixed(5)} : ${parseFloat(originlngLat.lng).toFixed(5)}` :
+      `${parseFloat(destinationlngLat.lat).toFixed(5)} : ${parseFloat(destinationlngLat.lng).toFixed(5)}`)
+    _self.directionPanel.searchDirectionByCoordinates(originlngLat, destinationlngLat)
+  }
```

Comment:
this implementation works ok, but has some drawbacks:
 * when moving an "origin" marker, the destination POI (id, name etc) will be lost in the URL (and vice versa)
 * it duplicates the implementation of `UrlPoi` label

It should be possible to use `directionPanel.selectOrigin` and `directionPanel.selectDestination` :
```js
this.directionPanel.selectOrigin(new UrlPoi(...))
```

Also note that `UrlPoi` is quite a bad name as it basically represents a POI that is initialized with a latitude and a longitude only, and it could be a good opportunity to rename it to something more explicit. `LatLonPoi` ?

Reply:
I made some changes according to your review @amatissart.

The `UrlPoi` has been renamed to `LatLonPoi` to be more significant.

The only dragged marker is now updated with its own coordinates as it's been wanted by his default behavior.

The use of `selectOrigin` and `selectDestination` has been implemented and made the custom `searchDirectionByCoordinates` implementation useless (removed now).

Reply:
Good. To make it even cleaner, we can use `latlon_poi.getInputValue()` instead of explicit strings here.

Also, `searchDirection()` is already called by `selectOrigin` or `selectDestination` and should be removed here. It currently triggers a extra request to the directions api.